saleor/graphql/product/filters.py

```diff
@@ -451,6 +452,46 @@ def _filter_products_is_published(qs, _, value, channel_slug):
     )
 
 
+def _filter_products_is_available(qs, _, value, channel_slug):
+    channel = Channel.objects.filter(slug=channel_slug).values("pk")
+    now = datetime.datetime.now(pytz.UTC)
+    if value:
+        product_channel_listings = ProductChannelListing.objects.filter(
+            Exists(channel.filter(pk=OuterRef("channel_id"))),
+            available_for_purchase_at__lte=now,
+        ).values("product_id")
+    else:
+        product_channel_listings = ProductChannelListing.objects.filter(
+            Exists(channel.filter(pk=OuterRef("channel_id"))),
+            Q(available_for_purchase_at__gt=now)
+            | Q(available_for_purchase_at__isnull=True),
+        ).values("product_id")
+
+    return qs.filter(Exists(product_channel_listings.filter(product_id=OuterRef("pk"))))
+
+
+def _filter_products_channel_field_from_date(qs, _, value, channel_slug, field):
+    channel = Channel.objects.filter(slug=channel_slug).values("pk")
```

Comment:
You should handle the empty value here too. And please add test for this scenario